### ISSUE TITLE
[Bug]: Cursor jumps when deleting a character in the default Python example

### ISSUE THREAD

**Issue by alexeyinkin** [apache/beam#25279]
### What happened?

Flutter 3.7.1

`/?path=SDK_PYTHON%2FPRECOMPILED_OBJECT_TYPE_EXAMPLE%2FWordCountWithMetrics&sdk=python`

Press Delete:

https://user-images.githubusercontent.com/44893228/216327660-572da874-bcfd-470a-baf5-5ace4ef7a8cf.mp4

### Issue Priority

Priority: 3 (minor)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [X] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by alexeyinkin** [comment]
.take-issue

**Comment by yescorp** [comment]
Foldable block that is folded is no longer recognizable because of the unclosed multiline string. 
Therefore, when we update the code inside `CodeController.setValue`, it is not the same as the text that is being edited.

In `CodeController.setValue` we have:
```dart
_updateCodeIfChanged(editResult.fullTextAfter);

if (newValue.text != _code.visibleText) {
  // Manually typed in a text that has become a hidden range.
  newValue = newValue.replacedText(_code.visibleText);
}
```

Which changes selection according to 

```dart
TextEditingValue replacedText(String newText) {
    if (newText == text) {
      return this;
    }

    final rangeAfter = newText.getChangedRange(
      text,
      attributeChangeTo: TextAffinity.upstream,
    );

    return TextEditingValue(
      text: newText,
      selection: TextSelection.collapsed(offset: rangeAfter.start),
    );
  }
```

It seems useful in some cases, because just changing this led to failing test cases.

**Comment by yescorp** [comment]
This behavior can be reproduced in different ways. But the root cause is the same.
- See https://github.com/akvelon/flutter-code-editor/issues/131

**Comment by alexeyinkin** [comment]
Fixed in flutter_code_editor 0.2.10.
